initial

trapacska opened the pull request:
https://github.com/bitrise-io/appcenter/pull/1/files/9bec804f39065b59757b14aaa6f04a1bca97bd1c..7eae4a1faf7dbda8b9d125fa8f7d1f109879808a

- extended base model
- renamed two methods
- added one new method

trapacska committed:
initial

lpusok commented on app.go:
```diff
@@ -0,0 +1,183 @@
+func (a App) NewRelease(filePath string, opts ...ReleaseOptions) (Release, error) {
+	// send file upload request
+	var (
+		postURL      = fmt.Sprintf("%s/v0.1/apps/%s/%s/release_uploads", baseURL, a.owner, a.name)
```
Is URL encoding needed for params? url.URL can also be used to construct an URL.

trapacska replied:
it is done by newrequest

lpusok commented on app.go:
```diff
@@ -0,0 +1,183 @@
+	owner, name string
+}
+
+// NewRelease ...
```
Can you copy some docs here or write short sentence.

lpusok commented on common.go:
```diff
@@ -0,0 +1,113 @@
+		return -1, err
+	}
+
+	uploadReq.Header.Set("Content-Type", w.FormDataContentType())
```
Does it work with every type of aab, apk, ipa? had some Content-type issues in deploy-to-bitrise-step.

trapacska replied:
yes

lpusok commented on common.go:
```diff
@@ -0,0 +1,113 @@
+	baseURL = `https://api.appcenter.ms`
+)
+
+func (c Client) jsonRequest(method, url string, body interface{}, response interface{}) (int, error) {
```
Is some kind of retry logic possible?

trapacska replied:
retry logic sounds like a requirement for client side implementation

lpusok commented on common.go:
```diff
@@ -0,0 +1,113 @@
+			return -1, err
+		}
+
+		if _, err = io.Copy(fw, f); err != nil {
```
Missing closing of f.

trapacska committed:
wired in model

trapacska committed:
pr updates

trapacska committed:
additional changes

trapacska committed:
added testers

trapacska committed:
reorder

trapacska committed:
regroup

trapacska committed:
reorder

trapacska committed:
statuscode fix

trapacska committed:
statuscode fix

trapacska committed:
changed filepath

trapacska committed:
added customizable headers to the client

trapacska committed:
added uploadFile

trapacska committed:
added content length

trapacska committed:
changed statuscode check

trapacska commented:
- replaced functions, regrouped them to have the methods on the same place as their root model
- added new function: uploadFile

https://github.com/bitrise-io/appcenter/pull/1/files/6869bf56fceb16f8be719248bc166114264ea5c2..18e09ac3a1d41011bbea4fc0f7e1fd7f2b3ac936

trapacska committed:
removed unused stuff
